pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -77,14 +68,30 @@ func (info *ClaimInfo) addCDIDevices(pluginName string, cdiDevices []string) err
 
 // annotationsAsList returns container annotations as a single list.
 func (info *ClaimInfo) annotationsAsList() []kubecontainer.Annotation {
-	info.RLock()
-	defer info.RUnlock()
-
-	var lst []kubecontainer.Annotation
+	var annotations []kubecontainer.Annotation
 	for _, v := range info.annotations {
-		lst = append(lst, v...)
+		annotations = append(annotations, v...)
+	}
+	return annotations
+}
+
+func (info *ClaimInfo) cdiDevicesAsList() []kubecontainer.CDIDevice {
+	var cdiDevices []kubecontainer.CDIDevice
+	for _, devices := range info.CDIDevices {
+		for _, device := range devices {
+			cdiDevices = append(cdiDevices, kubecontainer.CDIDevice{Name: device})
+		}
+	}
+	return cdiDevices
+}
+
+func (info *ClaimInfo) deepCopy() ClaimInfo {
```

Comment:
By convention, deep copy normally returns a pointer to the new object. We should actually probably rely on https://pkg.go.dev/k8s.io/gengo/examples/deepcopy-gen to generate them for us instead of writing all of the ones in this PR on our own.

Reply:
Great idea, thank you! done.